### PR TITLE
fix: Skip missing blocks in manual filtering

### LINKED ISSUE #194: [bug] Missing block numbers cause historical processing to fail
When historical processing does manual filtering of the past hour or two since the last index run, missing block numbers cause it to fail.

### PR DESCRIPTION
**Pull request by gabehamilton**
Proceed with historical filtering when missing block numbers are encountered.

### CODE REVIEW

File: indexer/queryapi_coordinator/src/historical_block_processing.rs [hunk 1]
```diff
@@ -330,7 +330,18 @@ async fn filter_matching_unindexed_blocks_from_lake(
     for current_block in (last_indexed_block + 1)..ending_block_height {
         // fetch block file from S3
         let key = format!("{}/block.json", normalize_block_height(current_block));
-        let block = s3::fetch_text_file_from_s3(&lake_bucket, key, s3_client.clone()).await?;
+        let block = s3::fetch_text_file_from_s3(&lake_bucket, key, s3_client.clone()).await;
+        if block.is_err() {
```
Comment (morgsmccauley) [inline indexer/queryapi_coordinator/src/historical_block_processing.rs:334]: What if it fails for reasons other than 'missing'? should we inspect the error message/status?

File: indexer/queryapi_coordinator/src/historical_block_processing.rs [hunk 2]
```diff
@@ -330,7 +330,18 @@ async fn filter_matching_unindexed_blocks_from_lake(
...
+            tracing::error!(
```
Comment (darunrs) [inline indexer/queryapi_coordinator/src/historical_block_processing.rs:335]: We may want to include the error itself in the trace or at least log it to the machine itself.

**Review by darunrs — APPROVED** [review]
LGTM. One comment on error logging.

File: indexer/queryapi_coordinator/src/historical_block_processing.rs (hunk 1 above)
Comment (gabehamilton) [reply indexer/queryapi_coordinator/src/historical_block_processing.rs]: Fair enough. New commit added with more specific error handling.

File: indexer/queryapi_coordinator/src/historical_block_processing.rs (hunk 2 above)
Comment (gabehamilton) [reply indexer/queryapi_coordinator/src/historical_block_processing.rs]: Just changed it to only catch the 404

**Review by morgsmccauley — APPROVED** [review]
Nice!

**Pull request merged**